src/sparcscore/pipeline/workflows.py

```diff
@@ -725,14 +725,31 @@ def cellpose_segmentation(self, input_image):
             else:
                 continue
 
+        # check if there are any cytosol masks that are assigned to multiple nuclei
+        unique_pairs = list(set(nucleus_cytosol_pairs.values()))
+        # check if there are any duplicate values
+        if len(unique_pairs) != len(nucleus_cytosol_pairs):
+            # if there are duplicate values, set them to 0
+            seen_cytosol = set()
+            for nucleus_id, cytosol_id in nucleus_cytosol_pairs.items():
+                if cytosol_id in seen_cytosol:
+                    nucleus_cytosol_pairs[nucleus_id] = 0
+                else:
+                    seen_cytosol.add(cytosol_id)
+
         # now we have all the nucleus cytosol pairs we can filter the masks
+        updated_cytosol_mask = np.zeros_like(masks_cytosol, dtype=bool)
+
         for nucleus_id, cytosol_id in nucleus_cytosol_pairs.items():
             if cytosol_id == 0:
-                masks_nucleus = np.where(masks_nucleus == nucleus_id, 0, masks_nucleus)
-                masks_cytosol = np.where(masks_cytosol == cytosol_id, 0, masks_cytosol)
+                masks_nucleus = np.where(masks_nucleus == nucleus_id, 0, masks_nucleus)  # set the nucleus to 0
+                masks_cytosol = np.where(masks_cytosol == cytosol_id, 0, masks_cytosol)  # set the cytosol to 0
```

Comment:
this second filtering does not need to be done. The cytosol id we are checking for is already 0 and we are setting it to 0.